### PR TITLE
feat(ci): use job matrix builds over synchronous bash script

### PR DESCRIPTION
**Pull request by getchoo**
bash script bad :bonk:

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -4,11 +4,24 @@ on:
   push:
     branches: ["main"]
   pull_request:
-    branches: ["main"]
```
Comment (ryanccn) [inline .github/workflows/build.yml:7]: why remove this :(

Comment (getchoo) [reply .github/workflows/build.yml:7]: it's pointless :trollface:

Comment (ryanccn) [reply .github/workflows/build.yml:7]: eh ok

**Review by ryanccn — APPROVED** [review]

**Comment by ryanccn** [comment]
ummm what happened to windows

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -4,11 +4,24 @@ on:
...
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        go_arch: [amd64, arm64]
+        include:
+          - os: ubuntu-latest
+            go_os: "linux"
+          - os: macos-latest
+            go_os: "darwin"
+          - os: windows-latest
+            go_os: "window"
```
Comment (ryanccn) [inline .github/workflows/build.yml:22]: window

**Review by ryanccn — CHANGES_REQUESTED** [review]

**Review by ryanccn — APPROVED** [review]

**Pull request merged**